git-secrets

```diff
@@ -106,6 +110,48 @@ scan_history() {
   process_output $? "${output}"
 }
 
+# Takes an argument like commitHashA..commitHashB
+# Scans modified files in each commit at each point in history
+scan_between() {
+  # Fail if unstaged changes exist
+  git update-index --refresh
+  [[ -n $(git status --porcelain=v1 2> /dev/null) ]] &&
+    printf "Commit or stash changes first." >&2 && exit 1
+
+  # Require scan range (HEAD/head or SHA1 hash)
+  # Abbreviated hashes minimum 7 chars allowed
+  local -r scan_range="${1}"
+  head_or_sha1="(HEAD|head|[0-9a-f]{7,40})(\^[0-9]{1})?"
+  [[ ! ${scan_range} =~ ^${head_or_sha1}[\.]{2,3}${head_or_sha1}$ ]] &&
+    printf "Must specify a valid scan range, i.e. 'gitHashA..gitHashB'" >&2 && exit 1
+
+  # No patterns means nothing to scan for (OK)
+  local combined_patterns=$(load_combined_patterns)
+  [ -z "${combined_patterns}" ] && return 0
+
+  # No commits means nothing to scan (OK)
+  local -r commits_cmd="git log ${scan_range} --format=format:%h --reverse"
+  IFS=$'\n' read -r -d '' -a commits < <( ${commits_cmd} && printf '\0' )
+  [ -z "${commits}" ] && return 0
+  unset IFS
+
+  local -r current_branch=$(git branch --show-current)
+
+  declare found
+  for commit_hash in "${commits[@]}"; do
+    git checkout "${commit_hash}" >& /dev/null
+    local prev_commit_hash=$(git rev-parse "${commit_hash}^1")
+    local files_changed=$(git diff "${prev_commit_hash}..${commit_hash}" --name-only)
+    local buff=$(GREP_OPTIONS= LC_ALL=C git grep -nwHEIP \ "${combined_patterns}" -- ${files_changed} | sed "s/^/${commit_hash}:/")
+    [[ -n ${buff} ]] && found+=$(echo -e "\r\n${buff}")
+    unset buff
+  done
+
+  git switch "${current_branch}" >& /dev/null
```

Comment:
Fixes detached head resulting from checkout. Assumes HEAD of whatever branch is checked out. Potentially not a great idea.

Reply:
I have solved this, stand by shortly as I will push the change.